MyModels.py

```diff
@@ -0,0 +1,158 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+    
+#     def find_index(self,entry):
+#         for i, item in self.list:
+#             if item==entry:
+#                 return i
+            
+#     def find_val(self,index):
+#         return self.list[index]
+
+#####################
+
+    def find_index(self,id):
+        return self.id_index[id]
+    
+    def find_val(self,id):
+        index = self.find_index(id)
+        return self.list[index]
+
+
+###################
+    def GetAll(self):
+        return self.list
+    
+    def delete(self, entry):
```

Comment:
Usually when you wanna delete an entry, you only have its `id`, not a memory reference to its full entry. Therefore, a `delete` method that takes an id for deleting is much more desirable than this.

Reply:
same problem due to simplifying ! I assure you BaseModel will have great changes in the very next commit.